src/ucp/core/ucp_worker.c

```diff
@@ -2721,10 +2731,17 @@ ucp_worker_discard_tl_uct_ep(ucp_worker_h worker, uct_ep_h uct_ep,
     req->send.discard_uct_ep.ucp_worker     = worker;
     req->send.discard_uct_ep.uct_ep         = uct_ep;
     req->send.discard_uct_ep.ep_flush_flags = ep_flush_flags;
-    uct_worker_progress_register_safe(worker->uct,
-                                      ucp_worker_discard_uct_ep_progress,
-                                      req, UCS_CALLBACKQ_FLAG_ONESHOT,
-                                      &cb_id);    
+
+    status = uct_ep_flush(uct_ep, req->send.discard_uct_ep.ep_flush_flags,
+                          &req->send.state.uct_comp);
+    if (status == UCS_OK) {
+        ucp_worker_discard_uct_ep_destroy_progress(req);
+    } else if (status != UCS_INPROGRESS) {
+        uct_worker_progress_register_safe(worker->uct,
+                                          ucp_worker_discard_uct_ep_progress,
+                                          req, UCS_CALLBACKQ_FLAG_ONESHOT,
+                                          &cb_id);
+    }
 }
```

Comment:
maybe just call "ucp_worker_discard_uct_ep_progress" and if returns 0 then schedule it?

Reply:
or maybe no need to schedule it, since ucp_worker_discard_uct_ep_progress would already schedule itself..

Reply:
done